torch/csrc/jit/passes/onnx/constant_map.cpp

```diff
@@ -0,0 +1,177 @@
+c10::optional<c10::SymbolicShape> ConstantValueMap::GetShape(
+    const std::string& tensorName) {
+  if (!HasShape(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().shapeMap[tensorName];
+}
+
+void ConstantValueMap::SetValue(
+    const std::string& tensorName,
+    const at::Tensor& value) {
+  ConstantValueMap::getInstance().tensorValueMap.emplace(tensorName, value);
+}
+
+bool ConstantValueMap::HasValue(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().tensorValueMap.find(tensorName) !=
+      ConstantValueMap::getInstance().tensorValueMap.end();
+}
+
+c10::optional<at::Tensor> ConstantValueMap::GetValue(
+    const std::string& tensorName) {
+  if (!HasValue(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().tensorValueMap[tensorName];
+}
+
+c10::optional<std::vector<int64_t>> ConstantValueMap::GetShapeInto1DInt64Vector(
+    const std::string& value_name) {
+  if (ConstantValueMap::HasShape(value_name)) {
+    auto shape_size = ConstantValueMap::GetShape(value_name).value();
+    std::vector<int64_t> shape_value;
+    if (shape_size.isComplete()) {
+      auto shape_symbol_list = shape_size.sizes().value();
+      for (const auto& v : shape_symbol_list) {
+        shape_value.emplace_back(v.static_size());
+      }
+      return c10::optional<std::vector<int64_t>>(shape_value);
+    }
+  }
+  return c10::nullopt;
+}
+
+c10::optional<std::vector<int64_t>> ConstantValueMap::
+    GetShapeInto1DInt64VectorWithOneUnknown(const std::string& value_name) {
+  if (ConstantValueMap::HasShape(value_name)) {
+    auto shape_size = ConstantValueMap::GetShape(value_name).value();
+    std::vector<int64_t> shape_value;
+    if (shape_size.isComplete()) {
+      auto shape_symbol_list = shape_size.sizes().value();
+      for (const auto& v : shape_symbol_list) {
+        shape_value.emplace_back(v.static_size());
+      }
+      return c10::optional<std::vector<int64_t>>(shape_value);
+    } else {
+      size_t count_unknown = 0;
+      auto shape_size_sizes = shape_size.sizes();
+      if (shape_size_sizes.has_value()) {
+        auto shape_symbol_list = shape_size_sizes.value();
+        for (const auto& v : shape_symbol_list) {
+          if (v.is_static()) {
+            shape_value.emplace_back(v.static_size());
+          } else {
+            shape_value.emplace_back(-1);
+            count_unknown += 1;
+          }
+        }
+        if (count_unknown == 1) {
+          return c10::optional<std::vector<int64_t>>(shape_value);
+        }
+      }
+    }
+  }
+  return c10::nullopt;
+}
+
+// accessor<int64_t, 1> for 1DInt64 case.
+std::vector<int64_t> ConstantValueMap::GetValueInto1DInt64Vector(
+    const std::string& value_name) {
+  auto value_size = ConstantValueMap::GetValue(value_name).value();
+  auto value_size_int = value_size.toType(at::ScalarType::Long);
+  std::vector<int64_t> value_temp;
+  value_temp.reserve(value_size_int.size(0));
```

Comment:
Check if value is 1d tensor? 
Does this support scalar?

Reply:
There is a check `TORCH_CHECK(dim() == N, "TensorAccessor expected ", N, " dims but tensor has ", dim());` in `TensorAccessor<T,N> accessor() const&`. So we don't check it here.
No, it only supports 1d tensor. For scalar, use `item()`.